### PR TITLE
Update objdetect benchmark formatting

### PR DESCRIPTION
**Pull request by AleksandrPanov**
- add formatting with pandas, add Checker, refactoring
- add aruco DetectorParameters read/write
- add statistics*

Here is an example of the program output:
image

### CODE REVIEW

File: python_benchmarks/objdetect_benchmark/objdetect_benchmark.py [hunk 1]
```diff
@@ -192,7 +193,7 @@ def transform_image(self, image):
         self.col_offset = int(self.background_image.shape[1] * self.rel_center[1] - image.shape[1] / 2)
         background_image = np.copy(self.background_image)
         background_image[self.row_offset:self.row_offset + image.shape[0],
-        self.col_offset:self.col_offset + image.shape[1]] = image
+                         self.col_offset:self.col_offset + image.shape[1]] = image
```
Comment (AleksandrPanov) [inline python_benchmarks/objdetect_benchmark/objdetect_benchmark.py:206]: tab refactoring

File: python_benchmarks/objdetect_benchmark/objdetect_benchmark.py [hunk 2]
```diff
@@ -255,24 +256,20 @@ def show(self, wait_key=0):
 
 class BackGroundObject(SyntheticObject):
     def __init__(self, *, num_rows, num_cols, color=0):
-        self.image = np.zeros((num_rows, num_cols), dtype=np.uint8)+color
+        self.image = np.zeros((num_rows, num_cols), dtype=np.uint8) + color
```
Comment (AleksandrPanov) [inline python_benchmarks/objdetect_benchmark/objdetect_benchmark.py:269]: tab refactoring

File: python_benchmarks/objdetect_benchmark/objdetect_benchmark.py [hunk 3]
```diff
@@ -255,24 +256,20 @@ def show(self, wait_key=0):
...
 
     def show(self, wait_key=0):
         cv.imshow("BackGroundObject", self.image)
         cv.waitKey(wait_key)
 
 
-def checkerboard(shape):
-    return ((np.indices(shape).sum(axis=0) % 2) * 255).astype(dtype=np.uint8)
-
-
```
Comment (AleksandrPanov) [inline python_benchmarks/objdetect_benchmark/objdetect_benchmark.py:268]: removed unused code

File: python_benchmarks/objdetect_benchmark/objdetect_benchmark.py [hunk 4]
```diff
@@ -255,24 +256,20 @@ def show(self, wait_key=0):
...
 class SyntheticAruco(SyntheticObject):
     def __get_size(self, cell_img_size):
         board_image_size = [0, 0]
         pix = cell_img_size / (1. + self.marker_separation)
         for i in range(0, 2):
-            board_image_size[i] = max(0, cell_img_size*(self.board_size[i]-2))
-            board_image_size[i] += pix*(1 + self.marker_separation/2)*min(2, self.board_size[i])
+            board_image_size[i] = max(0, cell_img_size * (self.board_size[i] - 2))
+            board_image_size[i] += pix * (1 + self.marker_separation / 2) * min(2, self.board_size[i])
```
Comment (AleksandrPanov) [inline python_benchmarks/objdetect_benchmark/objdetect_benchmark.py:282]: tab refactoring

File: python_benchmarks/objdetect_benchmark/objdetect_benchmark.py [hunk 5]
```diff
@@ -330,44 +327,70 @@ def read(self, path="test", filename="test"):
...
-            detected[int(marker_id)] = marker.reshape(4, 2)
-        for gold_id in gold_ids:
-            gold_corner = gold_corners[int(gold_id)]
-            if int(gold_id) in detected:
-                corner = detected[int(gold_id)]
-                loc_dist = get_norm(gold_corner, corner, type_dist)
-                if loc_dist < accuracy:
-                    dist += loc_dist
-                    detected_count += 1
-    return detected_count, total_count, dist
```
Comment (AleksandrPanov) [inline python_benchmarks/objdetect_benchmark/objdetect_benchmark.py:354]: move to `@staticmethod`

File: python_benchmarks/objdetect_benchmark/objdetect_benchmark.py [hunk 6]
```diff
@@ -330,44 +338,129 @@ def read(self, path="test", filename="test"):
...
+def set_plt():
+    # Turn interactive plotting off
+    plt.ioff()
+    plt.rcParams["figure.figsize"] = (15, 9)
+    plt.rcParams["figure.subplot.bottom"] = 0.3
+    plt.rcParams["figure.subplot.left"] = 0.05
+    plt.rcParams["figure.subplot.right"] = 0.99
+
+
+def print_category_frame(obj_type, category, statistics, accuracy, path):
```
Comment (asmorkalov) [inline python_benchmarks/objdetect_benchmark/objdetect_benchmark.py]: bad function name. Not clear what it does.

File: python_benchmarks/qr_codes/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+opencv-python==4.8.1.78
+packaging==23.2
+pandas==2.1.1
+Pillow==9.4.0
+pyparsing==3.1.1
+python-dateutil==2.8.2
+pytz==2023.3.post1
+scipy==1.11.3
+six==1.16.0
+tzdata==2023.3
```
Comment (asmorkalov) [inline python_benchmarks/qr_codes/requirements.txt]: The dependencies are to strict and some of them redundant. I propose to exclude numpy and opencv-python as the test version may differ from the requirements.

Comment (AleksandrPanov) [reply python_benchmarks/qr_codes/requirements.txt]: fixed

File: python_benchmarks/objdetect_benchmark/objdetect_benchmark.py (hunk 6 above)
Comment (AleksandrPanov) [reply python_benchmarks/objdetect_benchmark/objdetect_benchmark.py]: renamed to `get_and_print_category_statistic`

File: python_benchmarks/objdetect_benchmark/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,5 @@
+opencv-python
```
Comment (asmorkalov) [inline python_benchmarks/objdetect_benchmark/requirements.txt]: OpenCV-Python is redundant. I may test self-build OpenCV, but not the release one.

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**